rrdd/rrdd_server.ml

```diff
@@ -272,18 +274,25 @@ let forget_host_ds (ds_name : string) : unit =
     )
 
 let query_possible_dss rrdi =
-  let enabled_dss = Rrd.ds_names rrdi.rrd in
-  let open Ds in
-  let open Data_source in
-  List.map (fun ds -> {
-        name = ds.ds_name;
-        description = ds.ds_description;
-        enabled = List.mem ds.ds_name enabled_dss;
-        standard = ds.ds_default;
-        min = ds.ds_min;
-        max = ds.ds_max;
-        units = ds.ds_units;
-      }) rrdi.dss
+  rrdi.rrd.Rrd.rrd_dss
+  |> Array.map (fun ds ->
+    let enabled_source = List.find_opt (fun ds' -> ds'.Ds.ds_name = ds.Rrd.ds_name) rrdi.dss in
```

Comment:
How large is `rrdi.ds` - would it make sense to build a more efficient data structure rather than using linear search? I assume this would only have an impact with lists over 100 entries.

Reply:
A vanilla host has around 250 sources, a host with many VMs is likely to have thousands of datasources. Will use a Map instead.